packages/contracts/src/hardhat-deploy-ethers.ts

```diff
@@ -34,6 +35,23 @@ export const deployAndPostDeploy = async ({
   await hre.ethers.provider.waitForTransaction(result.transactionHash)
 
   if (result.newlyDeployed) {
+    if (!(await isHardhatNode(hre))) {
+      console.log(cyan('Verifying on Etherscan...'))
+      // Verification sometimes fails, even when the contract is correctly deployed and eventually
+      // verified. Possibly due to a race condition. We don't want to halt the whole deployment
```

Comment:
What's the error?

Reply:
Usually this, which happens on Kovan because the identical source code has already been verified. 
```
NomicLabsHardhatPluginError: The Etherscan API responded with a failure status.
The verification may still succeed but should be checked manually.
Reason: Already Verified
```

I could detect that particular error and throw on everything else, but IMO halting the deployment based on an external failure from the Etherscan API would be an overreaction. Catching any errors on this call seems like the best approach, as we can always go back and manually verify a single contract quite easily vs. restarting the process mid-deployment which feels more distruptive to me.

Reply:
Yeah that's fine. I'll just request that we open up an issue on GH to make this more robust in the future.